packages/intent-filler/src/strategies/basic.ts

```diff
@@ -0,0 +1,572 @@
+import { IPostRequest } from "hyperbridge-sdk"
+import { EVM_HOST } from "@/config/abis/EvmHost"
+import { viemClientFactory } from "@/config/client"
+import { privateKeyToAccount, privateKeyToAddress } from "viem/accounts"
+
+export class BasicFiller implements FillerStrategy {
+	name = "BasicFiller"
+	private privateKey: HexString
+
+	constructor(privateKey: HexString) {
+		this.privateKey = privateKey
+	}
+
+	/**
+	 * Determines if this strategy can fill the given order
+	 * @param order The order to check
+	 * @param config The filler configuration
+	 * @returns True if the strategy can fill the order
+	 */
+	async canFill(
+		order: Order,
+		config: FillerConfig,
+	): Promise<boolean> {
+		try {
+			const destClient = this.getPublicClient(order.destChain);
+			const sourceClient = this.getPublicClient(order.sourceChain)
+			const currentBlock = await destClient.getBlockNumber()
+			const deadline = BigInt(order.deadline)
+
+			if (deadline < currentBlock) {
+				console.debug(`Order expired at block ${deadline}, current block ${currentBlock}`)
+				return false
+			}
+
+			const isAlreadyFilled = await this.checkIfOrderFilled(order, sourceClient)
+			if (isAlreadyFilled) {
+				console.debug(`Order is already filled`)
+				return false
+			}
+
+			const hasEnoughTokens = await this.checkTokenBalances(order.outputs, destClient)
+			if (!hasEnoughTokens) {
+				console.debug(`Insufficient token balances for order`)
+				return false
+			}
+
+			const orderValue = await this.calculateOrderValue(order, destClient)
+			const requiredConfirmations = config.confirmationPolicy.getConfirmationBlocks(
+				chainIds[order.destChain as keyof typeof chainIds]!,
+				orderValue.toString(),
+			)
+			const sourceReceipt = await sourceClient.getTransactionReceipt({ hash: order.transactionHash })
+			const sourceConfirmations = await sourceClient.getTransactionConfirmations({
+				transactionReceipt: sourceReceipt,
+			})
+			if (sourceConfirmations < requiredConfirmations) {
+				console.debug(
+					`Insufficient confirmations for order, ${sourceConfirmations} confirmations, ${requiredConfirmations} required`,
+				)
+				return false
+			}
+
+			return true
+		} catch (error) {
+			console.error(`Error in canFill:`, error)
+			return false
+		}
+	}
+
+	/**
+	 * Calculates the expected profitability of filling this order
+	 * @param order The order to calculate profitability for
+	 * @returns The expected profit in a normalized unit (usually USD value or ETH equivalent)
+	 */
+	async calculateProfitability(
+		order: Order
+	): Promise<number> {
+		try {
+
+			const destClient = this.getPublicClient(order.destChain);
+			const sourceClient = this.getPublicClient(order.sourceChain)
+
+			const gasEstimateForFill = await this.estimateGasForFill(order, destClient)
+
+			const ethPriceUsd = await this.getEthPriceUsd(order, destClient)
+
+			const relayerFeeEth = parseEther("0.001") // Fixed fee in ETH, converted to wei
```

Comment:
how do I determine this?

Reply:
The relayer fee is the fee for redeeming the escrowed tokens on the source, so it should basically be at least a value that covers the `postGasEstimate`  gas with a buffer of at least 1%

Reply:
The relayer fee has to be profitable for any relayer that would try to deliver the message, so add a profit of 1-2% on top of the cost of executing the post request, you could also make this profit margin configurable.

Reply:
Okay thank you, fixed in the code.